metadata/src/main/java/org/apache/kafka/controller/ConfigurationControlManager.java

```diff
@@ -215,13 +215,17 @@ private void incrementalAlterConfigResource(ConfigResource configResource,
                     }
                     List<String> newValueParts = getParts(newValue, key, configResource);
                     if (opType == APPEND) {
-                        if (!newValueParts.contains(opValue)) {
-                            newValueParts.add(opValue);
+                        for (String value: opValue.split(",")) {
+                            if (!newValueParts.contains(value)) {
```

Comment:
Do we have an integration test which covers the case when the appended value is already present? It kind of looks like the zk path doesn't handle that.

By the way, I found the naming a little confusing here. I think `newValueParts` should be `currentValueParts` or something like that.

Reply:
Nice catch, found a small bug of `ConfigAdminManager` when working on this integration test.

for the variable naming, I think both are OK here, I used `oldValueList` to be a better one.